src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -223,6 +223,10 @@ public static void Clear()
             if (!Console.IsOutputRedirected)
             {
                 WriteStdoutAnsiString(TerminalFormatStringsInstance.Clear);
+                if (!string.IsNullOrEmpty(TerminalFormatStringsInstance.ClearScrollbackBuffer))
```

Comment:
according to https://unix.stackexchange.com/a/375784 we should do E3 before clear:

> If the terminfo/termcap entry has an E3 capability it, it first writes out that. 
> It then uses the clear capability to clear the visible screen.

Reply:
The snippet of `clear` shared in the issue had E3 being issued after clear, but I'm not sure it matters either way. I'll switch the order.

Reply:
I assume we manually validated the behavior on at least some system with the E3 and then clear?

Reply:
@stephentoub I had tested before, but unfortunately just assumed the switched order was okay per the linked answer. I just ran my manual test again and having E3 issued before clear does not clear the scrollback buffer. 

![image](https://github.com/dotnet/runtime/assets/3506334/42ca67d1-e003-49aa-8e1d-a3f1f90d5af9)
![image](https://github.com/dotnet/runtime/assets/3506334/6ec500d6-170c-423a-98be-909c6100fd30)

That means this order needs reversed back to the original: Clear, then E3. Would you mind making that change on your PR? Or do you want me to submit a new PR with the optimization mentioned below + this fix?

Reply:
> Would you mind making that change on your PR?

No problem; I switched the order back.